src/main/webapp/resources/js/pages/projects/share/index.js

```diff
@@ -9,28 +14,83 @@ import { render } from "react-dom";
  * @constructor
  */
 function ShareSamples() {
+  const dispatch = useDispatch();
   const [samples, setSamples] = React.useState();
+  const { originalSamples, currentProject, projectId } = useSelector(
+    (state) => state.shareReducer
+  );
+
+  const {
+    data: projects,
+    isLoading: projectLoading,
+  } = useGetProjectsToShareToQuery(currentProject, {
+    skip: !currentProject,
+  });
 
-  /*
-  CURRENTLY THIS IS JUST TO HAVE SOMETHING ON THE PAGE AND SHOW THAT THE SAMPLES
-  GET HERE.  DON'T WORRY TOO MUCH ABOUT LOOKING THROUGH THIS YET.
-   */
+  const { data: sampleIds } = useGetSampleIdsForProjectQuery(projectId, {
+    skip: !projectId,
+  });
+
+  React.useEffect(() => {
+    setSamples(originalSamples);
+  }, [originalSamples]);
 
   React.useEffect(() => {
-    const stringData = window.sessionStorage.getItem("share");
-    const data = JSON.parse(stringData);
+    if (sampleIds) {
+      setSamples(
+        originalSamples.map((sample) => ({
+          ...sample,
+          exists: sampleIds.includes(sample.id),
+        }))
+      );
+    }
+  }, [originalSamples, sampleIds]);
 
-    setSamples(data.samples);
-  }, []);
+  const updateCurrentSampleIds = (projectId) => dispatch(setProject(projectId));
 
   return (
-    <Table
-      loading={!samples}
-      dataSource={samples}
-      rowKey={(sample) => `sample-${sample.id}`}
-      columns={[{ title: "Name", dataIndex: "name" }]}
-    />
+    <Form layout="vertical">
+      <Space direction="vertical" style={{ display: "block" }} size="large">
+        <Form.Item label={i18n("ShareSamples.projects")}>
+          <Select
+            style={{ width: `100%` }}
+            loading={projectLoading}
+            options={projects?.map((project) => ({
+              label: project.name,
+              value: project.identifier,
+            }))}
+            onChange={updateCurrentSampleIds}
+          />
+        </Form.Item>
+        <Table
+          loading={!originalSamples}
+          dataSource={samples}
+          rowKey={(sample) => `sample-${sample.id}`}
+          columns={[
+            {
+              title: "Name",
```

Comment:
Internationalize

Reply:
This is already changed to a list in it's own component in the next PR

Reply:
Sounds good